lightning/src/chain/onchaintx.rs

```diff
@@ -376,9 +402,17 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 
 		// Try to aggregate outputs if their timelock expiration isn't imminent (package timelock
 		// <= CLTV_SHARED_CLAIM_BUFFER) and they don't require an immediate nLockTime (aggregable).
-		for req in requests {
+		'req_walk: for req in requests {
 			// Don't claim a outpoint twice that would be bad for privacy and may uselessly lock a CPFP input for a while
-			if let Some(_) = self.claimable_outpoints.get(req.outpoints()[0]) { log_trace!(logger, "Bouncing off outpoint {}:{}, already registered its claiming request", req.outpoints()[0].txid, req.outpoints()[0].vout); } else {
+			if let Some(_) = self.claimable_outpoints.get(req.outpoints()[0]) {
+				log_trace!(logger, "Bouncing off outpoint {}:{}, already registered its claiming request", req.outpoints()[0].txid, req.outpoints()[0].vout);
+			} else {
+				for locked_package in self.locktimed_packages.iter().map(|v| v.1.iter()).flatten() {
+					if locked_package.outpoints() == req.outpoints() {
+						continue 'req_walk;
+					}
+				}
```

Comment:
Hmm, I'm not really sure this is at all more readable, but maybe I just fine long chains of iterators a bit harder to read. Thoughts?

```
				let locked_package_timelock = self.locktimed_packages.iter().map(|v| v.1.iter()).flatten()
					.find_map(|locked_package| if locked_package.outpoints() == req.outpoints() {
						Some(locked_package.package_timelock())
					} else { None });
				if let Some(timelock) = locked_package_timelock {
					log_trace!(logger, "Ignoring second claim for outpoint {}:{}, we already have one which we're waiting on a timelock at {} for.",
						req.outpoints()[0].txid, req.outpoints()[0].vout, timelock);
					continue;
				}
```

Reply:
Given `find` returns an `Option`, you can use that instead of `find_map` with one line:

```rust
.find(|locked_package| locked_package.outpoints() == req.outpoints())
```

and keep the `log_trace` as it was.

Reply:
Oh! I misread the docs for `find`. Took your suggestion.